providers/com/peerj/normalizer.py

```diff
@@ -0,0 +1,123 @@
+import arrow
+
+import dateparser
+
+from share.normalize import *
+from share.normalize.utils import format_doi_as_url
+
+class ISSN(Parser):
+    schema = 'Link'
+    url = ctx
+    type = Static('issn')
+
+class Email(Parser):
+    email = ctx
+
+class PersonEmail(Parser):
+    email = Delegate(Email, ctx)
+
+class Institution:
+    name = ctx.author_institution
+
+class Affiliation:
+    # The entity used here could be any of the entity subclasses (Institution, Publisher, Funder, Organization).
+    entity = Delegate(Institution, ctx)
+
+class Link(Parser):
+    url = ctx
+    type = RunPython('get_link_type', ctx)
+
+    def get_link_type(self, link):
+        if 'dx.doi.org' in link:
+            return 'doi'
+        if 'peerj.com' in link:
+            return 'provider'
+        return 'misc'
+
+class ThroughLinks(Parser):
+    link = Delegate(Link, ctx)
+
+class Tag(Parser):
+    name = ctx
+
+class ThroughTags(Parser):
+    tag = (Delegate(Tag), ctx)
+
+class Publisher(Parser):
+    name = ctx
+
+class Institution(Parser):
+    name = RunPython('get_author_institute', ctx)
+
+    def get_author_institute(self, context):
+        # read into a set and passed back to erase duplicates
+        if 'author_institution' in context:
+            inst_set = set(context['author_institution'])
```

Comment:
You're losing the order here.

```python
In [5]: x = ['string1', 'another string', 'repeat', 'repeat', 'another string']

In [6]: list(set(x))
Out[6]: ['another string', 'repeat', 'string1']
```

2 potentials solutions are:

Create a dictionary that is index: value
Reverse it
Order the keys by value
```python
 x = ['string1', 'another string', 'repeat', 'repeat', 'another string']

x = dict(x)

x = reverse(x)

sorted(x.items())
```

Create a set and then remove values as you iterate over it
```python
x = ['string1', 'another string', 'repeat', 'repeat', 'another string']

other_x = set(x)

[other_x.pop(s) for s in x if s in other_x]
```